packages/@lwc/integration-karma/helpers/test-utils.js

```diff
@@ -339,6 +339,10 @@ window.TestUtils = (function (lwc, jasmine, beforeAll) {
             customElementCallbackReactionErrorListener,
             true
         ),
+        toThrowCallbackReactionErrorEvenInSyntheticLifecycleMode: errorMatcherFactory(
+            windowErrorListener,
+            true
+        ),
```

Comment:
@jmsjtu This is kind of weird but I couldn't think of a better name for this 😂

Reply:
😆 how about `toThrowCallbackReactionErrorInAllDomModes`, what you have works for me too. 

We can always change it later if we come up with a better name!